rosbag2_transport/src/rosbag2_transport/topic_filter.cpp

```diff
@@ -167,6 +174,13 @@ bool TopicFilter::take_topic(
       return false;
     }
 
+    if (!record_options_.all_topics &&
+      !record_options_.topic_types.empty() &&
+      !type_in_list(topic_type, record_options_.topic_types))
+    {
+      return false;
+    }
+
```

Comment:
@fujitatomoya I haven't gotten to the latest changes after my first round of review. However, as regards your last question 
> i was thinking that, --topic-types should be applied with --all-topics to filter them only specified types.

The command line option `--all-topics` means **All** topics excluding services regardless of the other inclusive filters among exclusive filters.

With the current design, It will be expected that two fundamental and conceptual conditions shall be true.
1. At least one options out of --all, --all-topics, --all-services, --services, --topics, --topic-types or --regex must be used.
2. Only one option out of --all, --all-topics, --topics, --topic-types or --regex can be used.

Reply:
okay that sounds reasonable, and it has been that, probably it better will be. thanks for the comment!